src/clj/fluree/db/json_ld/vocab.cljc

```diff
@@ -372,3 +376,106 @@
             (assoc :pred pred-map)
             (update-with db t vocab-flakes)
             (add-pred-datatypes (filterv #(> (count %) 1) preds)))))))
+
+;; schema serialization
+(def ^:const serialized-pred-keys [:id :datatype :subclassOf :parentProps :childProps])
+(def ^:const serialized-pred-keys-reverse (reverse serialized-pred-keys))
+
+(defn schema-tuple
+  [pred-map]
+  (reduce
+   (fn [acc next-key]
+     (let [next-val (get pred-map next-key)]
+       (cond
+         (set? next-val)
+         (if (seq next-val) ;; non-empty?
+           (conj acc
+                 (mapv #(if (iri/sid? %)
+                          (iri/serialize-sid %)
+                          %)
+                       next-val))
+           (if (seq acc) ;; if 'acc' is still empty, keep it that if nothing to add
+             (conj acc nil)
+             acc))
+
+         (iri/sid? next-val)
+         (conj acc (iri/serialize-sid next-val))
+
+         (nil? next-val)
+         (if (empty? acc)
+           acc
+           (conj acc nil))
+
+         :else
+         (conj acc next-val))))
+   (list)
+   serialized-pred-keys-reverse))
+
+(defn serialize-schema
+  "Serializes the schema map to a semi-compact json form which can be stored
+  in the index root file, allowing fast reification of the schema map without
+  requiring database queries to do so."
+  [{:keys [t pred] :as _db-schema}]
+  (let [pred-keys (mapv name serialized-pred-keys)
+        pred-vals (->> pred
+                       (filter #(string? (key %))) ;; every pred map is duplicated for both keys iri, and sid - keep only 1
+                       vals
+                       (mapv schema-tuple))]
+    {"t"    t
+     "pred" {"keys" pred-keys
+             "vals" pred-vals}}))
+
+(defn deserialize-pred-tuple
+  "Takes list of keys and tuples containing values
+  and turns them into a map with the respective keys and tuples"
+  [namespace-codes positions pred-vals]
+  (let [base-map base-property-map
```

Comment:
I don't think we need this extra binding here. We can just use the `base-property-map` directly below while initializing the `loop`.

Reply:
I do this intentionally when I'm using a constant and the function is not pure. Because it is so infrequent to do that, this helps call out this is happening.

But, it comes at a tiny cost, and especially if it isn't helping achieve that goal it serves no purpose - so I'll go ahead and get it change.